%YAML:1.0
---

# 相机标定图片读取路径
ImagePath: '/home/zhangtianyi/test_ws/src/cubot_radar/data/data1/sample/calibrator/2021-09-27/image/*.bmp'

# 相机标定结果保存路径
CalibrateResultPath: '/home/zhangtianyi/test_ws/src/cubot_radar/data/data1/sample/calibrator/2021-09-27/result.txt'

# 相机的内参矩阵
CameraMatrix: !!opencv-matrix
  rows: 3
  cols: 3
  dt: d
#  data: [1300.960356671899, 0, 615.945518440162,
#         0, 1300.392392370094, 512.5233172050018,
#         0, 0, 1]
  data: [1627, 0, 444.2,
         0, 1616.6, 610,
         0, 0, 1]


# 相机的畸变系数向量
DistortionVector: !!opencv-matrix
  rows: 5
  cols: 1
  dt: d
#  data: [-0.1735431407026466, 1.106654770287432, -0.0004794196813004683, 0.002110534407609221, -3.125580970999869]
  data: [-0.3994, 0.9118, -1.3788, -0.0058, 0.0039]

# 用于世界坐标转像素坐标的相机内参
InternalMatrix: !!opencv-matrix
  rows: 3
  cols: 4
  dt: d
  data: [1627, 0, 444.2,0,
         0, 1616.6, 610, 0,
         0, 0, 1, 0]
#  data: [2862.857840418645, 0, 710.6405078010379,
#        0, 2860.753215085235, 493.8626230826524,
#        0, 0, 1]

# 相机与雷达的外参矩阵
ExternalMatrix: !!opencv-matrix
  rows: 4
  cols: 4
  dt: d
  # 用ceres计算的相机雷达外参
#  data: [0.0309658,  -0.999369,  0.0174242,  0.0248853,
#         -0.0183874,  -0.0179992,  -0.999669,  -0.0456155,
#         0.999351,  0.0306352,  -0.0189332,  0.0334199,
#         0,  0,  0,  1]


   # 用OpenCV的PnP方法计算的相机雷达外参
#  data: [ 0.01844479964511292, -0.9996639268669866, 0.01821600084891511, 0.08921729445163448,
#          -0.01981325507503684, -0.01858097565157024, -0.9996310230615931, -0.03733665526143482,
#          0.999633545000058, 0.01807707566814376, -0.02014931874543258, 0.001587075738669714,
#          0,            0,           0,           1 ]
  data: [0.428848,  -0.901283 , 0.0614697 , -1.65724,
           0.0143869,  -0.0612219 , -0.99802 , -0.17614,
           0.903262 , 0.428884 , -0.0132882 , 0.396994,
         0,                  0,                   0,                    1]

